src/main/java/opendcs/dao/DbObjectCache.java

```diff
@@ -84,14 +86,28 @@ public void put(DBT dbObj)
 	 *
 	 * @param key DbKey of the object to be removed.
 	 */
-	public void remove(DbKey key)
+	public synchronized void remove(DbKey key)
 	{
 		ObjWrapper ow = keyObjMap.get(key);
 		if (ow == null)
 			return;
-		nameObjMap.remove(
-			nameIsCaseSensitive ? ow.obj.getUniqueName() : ow.obj.getUniqueName().toUpperCase());
-		keyObjMap.remove(key);
+		try
+		{
+			nameObjMap.remove(
+				nameIsCaseSensitive ? ow.obj.getUniqueName() : ow.obj.getUniqueName().toUpperCase());
+		}
+		catch(NoSuchElementException ex)
+		{
+			// another thread already removed it.
```

Comment:
with the synchronized keyword, won't that also prevent another thread from removing?

Reply:
Possible. But will require much more analysis to verify.